src/main/app/common/paymentPlan.ts

```diff
@@ -0,0 +1,57 @@
+import * as moment from 'moment'
+import 'moment-precise-range-plugin'
+class PaymentPlan {
+  private numberOfInstalments: number
+
+  constructor (
+    totalAmount: number,
+    instalmentAmount: number,
+    private frequencyInWeeks: number) {
+    this.numberOfInstalments = totalAmount / instalmentAmount
+  }
+
+  getPaymentLength (): string {
+    const now = moment()
+    const lastPaymentDate = this.getLastPaymentDate(now)
+
+    const paymentDuration = (moment as any).preciseDiff(now, lastPaymentDate, true)
+    const { years, months, days } = paymentDuration
+    const paymentLength: Array<string> = []
+
+    if (years) {
+      paymentLength.push(this.pluralize(years, 'year'))
+    }
+
+    if (months) {
+      paymentLength.push(this.pluralize(months, 'month'))
+    }
+
+    if (days) {
+      const weeks = moment.duration(days, 'day').get('week')
+
+      if (weeks) {
+        paymentLength.push(this.pluralize(weeks, 'week'))
+      }
+    }
+
+    return paymentLength.join(' ')
+  }
+
+  getLastPaymentDate (fromDate: moment.Moment = moment()): moment.Moment {
```

Comment:
the reason for default is getLastPaymentDate is to used to show last payment date for defendant and claimant (not part of this story). For defendant, lastPaymentDate is always calculated from moment.now but for claimant it has to show depending on defendant journey. Because defendant always uses moment.now, i think we decided to use default.